libs/wire-subsystems/src/Wire/HashPassword.hs

```diff
@@ -1,18 +1,49 @@
+{-# LANGUAGE RecordWildCards #-}
 {-# LANGUAGE TemplateHaskell #-}
 
 module Wire.HashPassword where
 
+import Crypto.KDF.Argon2 qualified as Argon2
 import Data.Misc
 import Imports
 import Polysemy
+import Polysemy.Input
+import Util.Options (PasswordHashingOptions (..))
 import Wire.API.Password (Password)
 import Wire.API.Password qualified as Password
 
 data HashPassword m a where
-  HashPassword :: PlainTextPassword8 -> HashPassword m Password
+  HashPassword6 :: PlainTextPassword6 -> HashPassword m Password
```

Comment:
More readable and more semantic than arbitrarily / unsafely coercing between the two types, but I also did not want to make this polymorphic directly like the implementation, I think it makes easier to visualise where pwd8 and pwd6 are used in code.